test_parens.py

```diff
@@ -0,0 +1,37 @@
+from parens import balanceness
+
+def test_balanced():
+    # Simple case.
+    assert balanceness(u'()') is 0
+    # Extension of the simple case.
+    assert balanceness(u'((()))') is 0
+    # Empty string is considered 'balanced'
+    assert balanceness(u'') is 0
+
+def test_balanced_longer():
+    assert balanceness(u'()()') is 0
+    assert balanceness(u'()()(())') is 0
+    assert balanceness(u'()(()())') is 0
+    assert balanceness(u'()(())()') is 0
+
+
+def test_open():
+    assert balanceness(u'(') is 1
+    assert balanceness(u'(()') is 1
+
+
+def test_open_longer():
+    assert balanceness(u'(())(') is 1
+    assert balanceness(u'()()(') is 1
+
+
+def test_broken():
+    assert balanceness(u')') is -1
+    assert balanceness(u'())') is -1
+    assert balanceness(u')()') is -1
+
+
+def test_broken_longer():
+    assert balanceness(u')))(((') is -1
+    assert balanceness(u'((())))') is -1
+    assert balanceness(u'()())()') is -1
```

Comment:
You should think about splitting all of these tests up into individual functions - Have as few assert statements per function as possible!